ZkBrokerFailureDetector doesn't work on migrate_to_kafka_2_5 branch

The broker failure detector doesn't work properly. 

The root cause is:
https://github.com/linkedin/cruise-control/blob/ee3d0e1c368c7c145c4ef1950924666b2c0e10a7/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/ZKBrokerFailureDetector.java#L93

This scala-to-java converter is throwing exception. 


By adding custom logs, the exception is like this:

```
2022/08/03 16:13:31.690 ERROR [AbstractBrokerFailureDetector] [AnomalyDetector-6] [kafka-cruise-control] [] Broker failure detector received exception:
java.lang.NoClassDefFoundError: scala/jdk/javaapi/CollectionConverters
        at com.linkedin.kafka.cruisecontrol.detector.ZKBrokerFailureDetector.aliveBrokers(ZKBrokerFailureDetector.java:102) ~[cruise-control-2.5.98-SNAPSHOT.jar:?]
        at com.linkedin.kafka.cruisecontrol.detector.AbstractBrokerFailureDetector.detectBrokerFailures(AbstractBrokerFailureDetector.java:63) [cruise-control-2.5.98-SNAPSHOT.jar:?]
        at com.linkedin.kafka.cruisecontrol.detector.ZKBrokerFailureDetector.run(ZKBrokerFailureDetector.java:71) [cruise-control-2.5.98-SNAPSHOT.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.ClassNotFoundException: scala.jdk.javaapi.CollectionConverters
        at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]
        at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
        at org.eclipse.jetty.webapp.WebAppClassLoader.loadClass(WebAppClassLoader.java:565) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
        ... 9 more
```

It seems like the project is using the scala 13 compiler, but the actual scala bytecode is still scala 12.

--- a/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AbstractBrokerFailureDetector.java
+++ b/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AbstractBrokerFailureDetector.java
@@ -30,7 +30,7 @@ import static org.apache.commons.io.FileUtils.writeStringToFile;
  * This class detects broker failures.
  */
 public abstract class AbstractBrokerFailureDetector extends AbstractAnomalyDetector implements Runnable {
-  
+
   protected static final Logger LOG = LoggerFactory.getLogger(AbstractBrokerFailureDetector.class);
   public static final String FAILED_BROKERS_OBJECT_CONFIG = "failed.brokers.object";
   // Config to indicate whether detected broker failures are fixable or not.
@@ -71,7 +71,7 @@ public abstract class AbstractBrokerFailureDetector extends AbstractAnomalyDetec
         // Report the failures to anomaly detector to handle.
         reportBrokerFailures();
       }
-    } catch (Exception e) {
+    } catch (Throwable e) {
       LOG.warn("Broker failure detector received exception: ", e);
     } finally {
       LOG.debug("Broker failure detection finished.");

--- a/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/ZKBrokerFailureDetector.java
+++ b/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/ZKBrokerFailureDetector.java
@@ -21,7 +21,7 @@ import kafka.zk.BrokerIdsZNode;
 import kafka.zk.KafkaZkClient;
 import org.apache.zookeeper.client.ZKClientConfig;
 import kafka.zookeeper.ZNodeChildChangeHandler;
-import scala.jdk.javaapi.CollectionConverters;
+import scala.collection.JavaConverters;
 
 import static java.util.stream.Collectors.toSet;
 
@@ -90,7 +90,7 @@ public class ZKBrokerFailureDetector extends AbstractBrokerFailureDetector {
   @Override
   Set<Integer> aliveBrokers() {
     // We get the alive brokers from ZK directly.
-    return CollectionConverters.asJava(_kafkaZkClient.getAllBrokersInCluster()).stream().map(Broker::id).collect(toSet());
+    return JavaConverters.asJavaCollection(_kafkaZkClient.getAllBrokersInCluster()).stream().map(Broker::id).collect(toSet());
   }
 
   /**

--- a/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/ZKBrokerFailureDetector.java
+++ b/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/ZKBrokerFailureDetector.java
@@ -90,6 +90,8 @@ public class ZKBrokerFailureDetector extends AbstractBrokerFailureDetector {
   @Override
   Set<Integer> aliveBrokers() {
     // We get the alive brokers from ZK directly.
+    // The JavaConverters API is deprecated in scala 2.13.0, and the new API is CollectionConverter. However, LinkedIn is still using
+    //  scala 2.12, which will cause it fail if it uses CollectionConverter.
     return JavaConverters.asJavaCollection(_kafkaZkClient.getAllBrokersInCluster()).stream().map(Broker::id).collect(toSet());
   }
 

